Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,373 @@
+            " remain facing the camera while IsBeingPlaced is true.  If true, the object will rotate according to the surface normal" +
+            " if there is a hit.")]
+        private bool rotateAccordingToSurface = false;
+
+        /// <summary>
+        /// If false, the game object to place will not change its rotation according to the surface hit.  The object will
+        /// remain facing the camera while IsBeingPlaced is true.  If true, the object will rotate according to the surface normal
+        /// if there is a hit.
+        /// </summary>
+        public bool RotateAccordingToSurface
+        {
+            get => rotateAccordingToSurface;
+            set => rotateAccordingToSurface = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.")]
+        private LayerMask[] magneticSurfaces = { UnityEngine.Physics.DefaultRaycastLayers };
+
+        /// <summary>
+        /// Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.
+        /// </summary>
+        public LayerMask[] MagneticSurfaces
+        {
+            get => magneticSurfaces;
+            set => magneticSurfaces = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.")]
+        private bool debugEnabled = true;
+
+        /// <summary>
+        /// If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.
+        /// </summary>
+        public bool DebugEnabled
+        {
+            get => debugEnabled;
+            set => debugEnabled = value;
+        }
+
+        [SerializeField, FormerlySerializedAs("OnPlacingStarted")]
+        [Tooltip("This event is triggered once when the game object to place is selected.")]
+        private UnityEvent onPlacingStarted = new UnityEvent();
+
+        /// <summary>
+        /// This event is triggered once when the game object to place is selected.
+        /// </summary>
+        public UnityEvent OnPlacingStarted
+        {
+            get => onPlacingStarted;
+            set => onPlacingStarted = value;
+        }
+
+        [SerializeField, FormerlySerializedAs("OnPlacingStopped")]
+        [Tooltip("This event is triggered once when the game object to place is unselected, placed.")]
+        private UnityEvent onPlacingStopped = new UnityEvent();
+
+        /// <summary>
+        /// This event is triggered once when the game object to place is unselected, placed.
+        /// </summary>
+        public UnityEvent OnPlacingStopped
+        {
+            get => onPlacingStopped;
+            set => onPlacingStopped = value;
+        }
+
+        protected internal int GameObjectLayer { get; protected set; }
```

Comment:
Nit: Add a doccomment on what this is (since it's protected and could be extended by others)

Reply:
Added
```
        /// <summary>
        /// The current game object layer before it is temporarily switched to IgnoreRaycast while placing the game object.
        /// </summary>
        protected internal int GameObjectLayer { get; protected set; }
```